CRM/CiviAwards/BAO/AwardDetail.php

```diff
@@ -76,4 +77,42 @@ private static function validateDates(array $params) {
     }
   }
 
+
+  /**
+   * Validates review_field options to ensure legal options are passed.
+   *
+   * @param array $params
+   *  Request Parameters
+   *
+   * @throws \Exception
+   */
+  private static function validateReviewFields(array &$params) {
+    $reviewFieldsLegalOptions = [
+      'id' => FILTER_VALIDATE_INT,
+      'weight' => FILTER_VALIDATE_INT,
+      'required' => FILTER_VALIDATE_BOOLEAN
+    ];
+    $reviewFields = CRM_Utils_Array::value('review_fields', $params);
+    if (is_array($reviewFields)) {
+      foreach ($reviewFields as &$reviewFieldOptions) {
+        $reviewFieldOptions = filter_var_array($reviewFieldOptions, $reviewFieldsLegalOptions, FALSE);
+        if (count($reviewFieldOptions) !== count($reviewFieldsLegalOptions)) {
```

Comment:
Also if using the filter_var_array function, if a field fails validation, the field is still present, the value will be false. See: https://3v4l.org/nk4PS